### PR TITLE
Sensor Filter Type Checking Bug Fix

### LINKED ISSUE #27: Sensor Filter bound checks may fail
The bound checking logic in [sensor_filter.py](https://github.com/UBCSailbot/boat_controller/blob/master/python/sensor_filter.py) will break if, for example, `None` is passed and we expect to check the bounds of a `float`. The controller node will crash as a result.

Change line 30 to return `False` right away if the type checking fails

https://github.com/UBCSailbot/boat_controller/blob/f14bd2d82f6fd8fc13b2258a44fd2b0e2a9fe74a/python/sensor_filter.py#L30-L32

Add tests to [test_SensorFilter.py](https://github.com/UBCSailbot/boat_controller/blob/master/test/test_SensorFilter.py) according to this change.

### PR DESCRIPTION
**Pull request by DFriend01**
Resolves #27

### CODE REVIEW

**Review by patrick-5546 — CHANGES_REQUESTED** [review]
Generally looks good, maybe add `rospy.logwarn()` if `SensorFilter.filter()` is false to make it easy to debug. I don't think this should ever happen, but if it does that means something is wrong. Once you have made the changes I can test it on my computer with local pathfinding and mock sensor data.

File: test/test_SensorFilter.py [hunk 1]
```diff
@@ -25,6 +25,13 @@ def test_invalid_nonetype(self):
         expected_type = float
         self.assertFalse(SensorFilter.filter(reading, lowerbound, upperbound, expected_type))
 
+    def test_bad_type(self):
```
Comment (patrick-5546) [inline test/test_SensorFilter.py:28]: If the reading is an int do we want to not update the field? Would anything break if the reading was an int instead of a float? Maybe you could just cast it to a float.

File: test/test_SensorFilter.py [hunk 2]
```diff
@@ -25,6 +25,13 @@ def test_invalid_nonetype(self):
...
+        reading = 3
+        lowerbound = 1.0
+        upperbound = 5.0
+        expected_type = float
+        self.assertFalse(SensorFilter.filter(reading, lowerbound, upperbound, expected_type))
```
Comment (patrick-5546) [inline test/test_SensorFilter.py:33]: If the reading is an int do we want to not update the field? Would anything break if the reading was an int instead of a float? Maybe you could just cast it to a float.

Comment (patrick-5546) [reply test/test_SensorFilter.py:33]: I checked that there is nothing stopping a field from being the wrong type, for example, min_voltage is `float32` (https://github.com/UBCSailbot/sailbot-msg/blob/master/msg/min_voltage.msg) yet I can do this:

```console
$ python
Python 2.7.17 (default, Mar 18 2022, 13:21:42) 
[GCC 7.5.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from sailbot_msg.msg import min_voltage
>>> m = min_voltage(min_voltage=3)
>>> type(m)
<class 'sailbot_msg.msg._min_voltage.min_voltage'>
>>> type(m.min_voltage)
<type 'int'>

```

Comment (DFriend01) [reply test/test_SensorFilter.py:33]: This is a valuable corner case that I did not consider. I will look into resolving this.
